### PR TITLE
P2p preheat ui

### PR DESCRIPTION
**Pull request by kofj**
1. Instances page.
2. Histories page.
3. E2E test.

Signed-off-by: 疯魔慕薇 <kfanjian@gmail.com>

### CODE REVIEW

File: src/portal/src/app/distribution/distribution-history/distribution-history.component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    this.disService.getDistributionHistories(this.queryParam).subscribe(
+      response => {
+        this.totalCount = Number.parseInt(
+          response.headers.get('x-total-count')
+        );
+        this.records = response.body;
+      },
+      err => console.error(err)
+    );
+    this.loading = false;
```
Comment (jwangyangls) [inline src/portal/src/app/distribution/distribution-history/distribution-history.component.ts]: loading should be cleared after asynchronous processing is finished

File: src/portal/src/app/distribution/distribution-history/distribution-history.component.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    this.loading = true;
+    this.queryParam.page = this.currentPage;
+    this.disService.getDistributionHistories(this.queryParam).subscribe(
+      response => {
+        this.totalCount = Number.parseInt(
+          response.headers.get('x-total-count')
+        );
+        this.records = response.body;
+      },
+      err => console.error(err)
```
Comment (jwangyangls) [inline src/portal/src/app/distribution/distribution-history/distribution-history.component.ts]: show error in page and user should see it

File: src/portal/src/app/distribution/distribution-instances/distribution-instances.component.ts [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    this.disService.getInstances(this.queryParam).subscribe(
+      response => {
+        this.totalCount = Number.parseInt(
+          response.headers.get('x-total-count')
+        );
+        this.instances = response.body as DistributionInstance[];
+      },
+      err => this.msgHandler.error(err)
+    );
+    this.loading = false;
```
Comment (jwangyangls) [inline src/portal/src/app/distribution/distribution-instances/distribution-instances.component.ts]: the same as above

File: src/portal/src/app/distribution/distribution.module.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { DistributionHistoryComponent } from './distribution-history/distribution-history.component';
+import { DistributionInstancesComponent } from './distribution-instances/distribution-instances.component';
+// import { DistributionProvidersComponent } from './distribution-providers.bak/distribution-providers.component';
```
Comment (jwangyangls) [inline src/portal/src/app/distribution/distribution.module.ts]: remove

File: src/portal/src/i18n/lang/en-us-lang.json [hunk 5]
```diff
@@ -1373,5 +1382,66 @@
         "HELP_INFO_1": "The default scanner has been installed. To install other scanners refer to the ",
         "HELP_INFO_2": "documentation."
 
+    },
+    "DISTRIBUTION": {
```
Comment (jwangyangls) [inline src/portal/src/i18n/lang/en-us-lang.json]: This part needs to be added in other languages

**Comment by AllForNothing** [comment]
Will improve with subsequent PR.   Merge and close

**Pull request merged**